src/actions.rs

```diff
@@ -0,0 +1,886 @@
+            NullAction::Pause { duration } => WDActions::NullActionItem::General(
+                WDActions::GeneralAction::Pause(WDActions::PauseAction {
+                    duration: Some(duration),
+                }),
+            ),
+        }
+    }
+}
+
+/// An action performed with a keyboard.
+#[derive(Debug, Clone)]
+pub enum KeyAction {
+    /// Pause action.
+    /// Useful for adding pauses between other key actions.
+    Pause {
+        /// The pause duration, given in milliseconds.
+        duration: u64,
+    },
+    /// Key up action.
+    Up {
+        /// The key code, e.g. `'a'`. See the [`Keys`] module for special key codes.
+        ///
+        /// [`Keys`]: crate::keys::Keys
+        value: char,
+    },
+    /// Key down action.
+    Down {
+        /// The key code, e.g. `'a'`. See the [`Keys`] module for special key codes.
+        ///
+        /// [`Keys`]: crate::keys::Keys
+        value: char,
+    },
+}
+
+impl From<KeyAction> for WDActions::KeyActionItem {
+    fn from(ka: KeyAction) -> Self {
+        use webdriver::actions::{KeyAction as WDKeyAction, KeyDownAction, KeyUpAction};
+        match ka {
+            KeyAction::Pause { duration } => WDActions::KeyActionItem::General(
+                WDActions::GeneralAction::Pause(WDActions::PauseAction {
+                    duration: Some(duration),
+                }),
+            ),
+            KeyAction::Up { value } => {
+                WDActions::KeyActionItem::Key(WDKeyAction::Up(KeyUpAction {
+                    value: value.to_string(),
+                }))
+            }
+            KeyAction::Down { value } => {
+                WDActions::KeyActionItem::Key(WDKeyAction::Down(KeyDownAction {
+                    value: value.to_string(),
+                }))
+            }
+        }
+    }
+}
+
+/// An action performed with a pointer device. See `PointerActionType` for
+/// the availabledevice types.
+#[derive(Debug, Clone)]
+pub enum PointerAction {
+    /// Pause action.
+    /// Useful for adding pauses between other key actions.
+    Pause {
+        /// The pause duration, given in milliseconds.
+        duration: u64,
+    },
+    /// Pointer button down.
+    Down {
+        /// The mouse button index. Button values are as follows:
+        /// - Left = 0
+        /// - Middle = 1
+        /// - Right = 2
+        button: u64,
+    },
+    /// Pointer button up.
+    Up {
+        /// The mouse button index. Button values are as follows:
+        /// - Left = 0
+        /// - Middle = 1
+        /// - Right = 2
+        button: u64,
+    },
+    /// Pointer move action. Duration is specified in milliseconds (can be 0).
+    /// The x and y offsets are relative to the origin.
+    Move {
+        /// The move duration, given in milliseconds.
+        duration: u64,
+        /// The origin that the `x` and `y` coordinates are relative to.
+        origin: PointerOrigin,
+        /// `x` offset, in pixels.
+        x: i64,
```

Comment:
Do we know that pixel values must be integers (and not floats)?

Reply:
The `webdriver` type `PointerMoveAction` has them all as i64, so that's all we've got.

Reply:
Ah, awkward. But fair enough.